grammar.js

```diff
@@ -25,6 +25,13 @@ module.exports = grammar({
 
     expression: ($) => seq(ANYTHING, NEWLINE),
 
-    result: ($) => token(prec(-1, ANYTHING)),
+    result: ($) =>
+      prec.right(
+        seq(sep1(token(prec(-1, ANYTHING)), NEWLINE), optional(NEWLINE))
```

Comment:
Do we need the optional newline given that `$.source` allows them anywhere?

Reply:
I originally left that optional newline off but I think it needs to be included in the `prec.right(..)` or else the prec.right consumes the NEWLINE and thinks its missing the result_token1 (the `token(prec(-1, ANYTHING))`)

<details><summary>the tests failures I see without the `optional(NEWLINE)`</summary>

```
main:
    ✓ Empty prompt with parens
    ✓ Multiple empty prompts without parens
    ✓ Prompts with elixir expressions but no result
    ✗ Prompts with elixir expressions and results
    ✗ Prompts with elixir expressions and one final results
    ✓ A continued prompt
    ✗ Continued prompts and results
    ✗ Multi-line results
    ✓ Missing newline on a result

4 failures:

expected / actual

  1. Prompts with elixir expressions and results:

    (source
      (evaluation_block
        (prompt_line
          (prompt)
          (expression))
        (result
          (MISSING "result_token1"))))
        (result))
      (evaluation_block
        (prompt_line
          (prompt)
          (expression))
        (result)))


  2. Prompts with elixir expressions and one final results:

    (source
      (evaluation_block
        (prompt_line
          (prompt)
          (expression)))
      (evaluation_block
        (prompt_line
          (prompt)
          (expression))
        (result))
      (ERROR))
        (result)))


  3. Continued prompts and results:

    (source
      (evaluation_block
        (prompt_line
          (prompt)
          (expression))
        (prompt_line
          (prompt)
          (expression))
        (result
          (MISSING "result_token1"))))
        (result))
      (evaluation_block
        (prompt_line
          (prompt)
          (expression))
        (prompt_line
          (prompt)
          (expression))
        (result)))


  4. Multi-line results:

    (source
      (evaluation_block
        (prompt_line
          (prompt)
          (expression))
        (prompt_line
          (prompt)
          (expression))
        (prompt_line
          (prompt)
          (expression))
        (result
          (MISSING "result_token1"))))
        (result))
      (evaluation_block
        (prompt_line
          (prompt)
          (expression))
        (result)))
```

</details>

Reply:
Makes sense, looks good then!